src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -244,15 +251,32 @@ delay_t eval_delay(const bottomless_token_bucket& tb) noexcept {
     return delay_t(muldiv(-tb.tokens(), delay_t::period::den, tb.quota()));
 }
 
+std::optional<quota_t> optional_min(
+  const std::optional<quota_t> lhs, const std::optional<quota_t> rhs) {
+    if (lhs && rhs) {
+        return std::min(*lhs, *rhs);
+    } else if (lhs) {
+        return lhs;
+    } else if (rhs) {
+        return rhs;
+    } else {
+        return std::nullopt;
+    }
+}
+
 } // namespace
 
 ingress_egress_state<std::optional<quota_t>>
 snc_quota_manager::calc_node_quota_default() const {
     // here will be the code to merge node limit
     // and node share of cluster limit; so far it's node limit only
     const ingress_egress_state<std::optional<quota_t>> default_quota{
-      .in = _kafka_throughput_limit_node_bps.in(),
-      .eg = _kafka_throughput_limit_node_bps.eg()};
+      .in = optional_min(
+        _kafka_throughput_limit_node_bps.in(),
+        _kafka_throughput_limit_cluster_bps.in()),
```

Comment:
i would have assumed that instead of min/max the node-level control would override the cluster-level control. is there some reasoning behind this?

Reply:
So overriding cluster limit with node limit will work differently if cluster limit is set to a smaller value than node limit. A configuration like this does not make much sense. So I think it does not really matter whether it's minimum or override. No opinion from me here; if you think it should be an override instead I will change it - LMK.

Reply:
Let's leave it min for now. Generally though we have a hierarchy of configs. Cluster and node where you get more specific at the node level. That is far more easy to reason about than minimum which would require knowing two values to understand the expected outcome...

Reply:
I can agrgue a bit for the min approach here: with cluster wide limit implemented, it will make total sense to configure both node and cluster limit, and their interaction won't be hierarchial but actually more resembling (but not exactly being a) min function.